graylog2-web-interface/src/components/configurations/UrlWhiteListForm.jsx

```diff
@@ -0,0 +1,213 @@
+  disabled: boolean,
+  onUpdate: (config: WhiteListConfig, valid: boolean) => void
+};
+
+const UrlWhiteListForm = ({ urls, onUpdate, disabled }: Props) => {
+  const literal = 'literal';
+  const regex = 'regex';
+  const options = [{ value: literal, label: 'Literal' }, { value: regex, label: 'Regex' }];
+  // eslint-disable-next-line prefer-const
+  let inputs = {};
+  const [config, setConfig] = useState<WhiteListConfig>({ entries: urls, disabled });
+  const [validationState, setValidationState] = useState({ errors: [] });
+
+  const _onAdd = (event: Event) => {
+    event.preventDefault();
+    setConfig({ ...config, entries: [...config.entries, { id: uuid(), title: '', value: '', type: literal }] });
+  };
+
+  const _onRemove = (event: MouseEvent, idx: number) => {
+    event.preventDefault();
+    // eslint-disable-next-line prefer-const
+    let stateUpdate = ObjectUtils.clone(config);
+    const validationUpdate = ObjectUtils.clone(validationState);
+    validationUpdate.errors[idx] = null;
+    setValidationState(validationUpdate);
+    stateUpdate.entries.splice(idx, 1);
+    setConfig(stateUpdate);
+  };
+
+  const validURL = (str: string) => {
+    let isValid = true;
+    try {
+      // eslint-disable-next-line no-unused-vars
+      const test = new URL(str);
+    } catch (e) {
+      isValid = false;
+    }
+    return isValid;
+  };
+
+  const _isFormValid = (): boolean => {
+    let isValid = true;
+    if (validationState.errors.length > 0
+      && validationState.errors.find((el => (el && el.title && el.title.valid) === false
+      || (el && el.value && el.value.valid === false)))) {
+      isValid = false;
+    }
+    return isValid;
+  };
+
+  const _updateState = (idx: number, type: string, name: string, value: string) => {
+    const stateUpdate = ObjectUtils.clone(config);
+    stateUpdate.entries[idx][name] = value;
+    stateUpdate.entries[idx] = { ...stateUpdate.entries[idx], type };
+    setConfig(stateUpdate);
+  };
+
+  const _updateValidationError = (idx: number, type: string, name: string, result: Object, value: string) => {
+    const validationUpdate = ObjectUtils.clone(validationState);
+    validationUpdate.errors[idx] = { ...validationUpdate.errors[idx], [name]: result };
+    setValidationState(validationUpdate);
+    _updateState(idx, type, name, value);
+  };
+
+  const _validate = (name: string, idx: number, type: string, value: string): void => {
+    switch (name) {
+      case 'title': {
+        const result = value.trim().length <= 0 ? { valid: false } : { valid: true };
+        _updateValidationError(idx, type, name, result, value);
+      }
+        break;
+      case 'value':
+        if (type === literal) {
+          const result = validURL(value) ? { valid: true } : { valid: false };
+          _updateValidationError(idx, type, name, result, value);
+        } else if (type === regex && value.trim().length > 0) {
+          const promise = ToolsStore.testRegexValidity(value);
```

Comment:
I think we should use `lodash.debounce()` to avoid calling this API method on every key press.

Reply:
I agree !

Reply:
Yes agree, i added debounce.